Dockerfile update

Hi,

Can some clever person help fixing the Dockerfile? :)

Andre

Sure, sorry:

```
❯ docker build --platform linux/amd64 .
Sending build context to Docker daemon  839.2kB
Step 1/23 : FROM golang:1.17 AS devbuild
1.17: Pulling from library/golang
d836772a1c1f: Pull complete 
66a9e63c657a: Pull complete 
d1989b6e74cf: Pull complete 
c28818711e1e: Pull complete 
9d6246ba248c: Pull complete 
21d43f0d73c2: Pull complete 
d8a1c5873f40: Pull complete 
Digest: sha256:87262e4a4c7db56158a80a18fefdc4fee5accc41b59cde821e691d05541bbb18
Status: Downloaded newer image for golang:1.17
 ---> 742df529b073
Step 2/23 : ENV CGO_ENABLED=0
 ---> Running in 57df848d3402
Removing intermediate container 57df848d3402
 ---> 1b0d218f3967
Step 3/23 : WORKDIR /build
 ---> Running in 87b03544d57c
Removing intermediate container 87b03544d57c
 ---> 758af69856ec
Step 4/23 : COPY . ./
 ---> b91c7a79c0f9
Step 5/23 : RUN go build -o damon ./cmd/damon
 ---> Running in 460f377e9824
go: downloading github.com/gdamore/tcell/v2 v2.5.3
go: downloading github.com/rivo/tview v0.0.0-20220911190240-55965cf21d8e
go: downloading github.com/jessevdk/go-flags v1.5.0
go: downloading github.com/hashicorp/nomad/api v0.0.0-20210804153318-c67b69bd0cc6
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading golang.org/x/sys v0.0.0-20220909162455-aba9fc2a8ff2
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/hashicorp/cronexpr v1.1.1
go: downloading github.com/hashicorp/go-cleanhttp v0.5.2
go: downloading github.com/hashicorp/go-rootcerts v1.0.2
go: downloading github.com/mitchellh/mapstructure v1.4.1
go: downloading github.com/gdamore/encoding v1.0.0
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading golang.org/x/term v0.0.0-20220722155259-a9ba230a4035
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/rivo/uniseg v0.4.2
# github.com/rivo/uniseg
/go/pkg/mod/github.com/rivo/uniseg@v0.4.2/properties.go:137:6: missing function body
/go/pkg/mod/github.com/rivo/uniseg@v0.4.2/properties.go:137:20: syntax error: unexpected [, expecting (
note: module requires Go 1.18
The command '/bin/sh -c go build -o damon ./cmd/damon' returned a non-zero code: 2

```

Oh, sorry, just had to update Go to 1.18 in the Dockerfile for it to build. :)

Also sorry for not adding useful information to this issue. The previous error looked like this:

```
❯ docker build --platform linux/amd64 .
Sending build context to Docker daemon  839.2kB
Step 1/23 : FROM golang:1.17 AS devbuild
1.17: Pulling from library/golang
d836772a1c1f: Pull complete 
66a9e63c657a: Pull complete 
d1989b6e74cf: Pull complete 
c28818711e1e: Pull complete 
9d6246ba248c: Pull complete 
21d43f0d73c2: Pull complete 
d8a1c5873f40: Pull complete 
Digest: sha256:87262e4a4c7db56158a80a18fefdc4fee5accc41b59cde821e691d05541bbb18
Status: Downloaded newer image for golang:1.17
 ---> 742df529b073
Step 2/23 : ENV CGO_ENABLED=0
 ---> Running in 57df848d3402
Removing intermediate container 57df848d3402
 ---> 1b0d218f3967
Step 3/23 : WORKDIR /build
 ---> Running in 87b03544d57c
Removing intermediate container 87b03544d57c
 ---> 758af69856ec
Step 4/23 : COPY . ./
 ---> b91c7a79c0f9
Step 5/23 : RUN go build -o damon ./cmd/damon
 ---> Running in 460f377e9824
go: downloading github.com/gdamore/tcell/v2 v2.5.3
go: downloading github.com/rivo/tview v0.0.0-20220911190240-55965cf21d8e
go: downloading github.com/jessevdk/go-flags v1.5.0
go: downloading github.com/hashicorp/nomad/api v0.0.0-20210804153318-c67b69bd0cc6
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading golang.org/x/sys v0.0.0-20220909162455-aba9fc2a8ff2
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/hashicorp/cronexpr v1.1.1
go: downloading github.com/hashicorp/go-cleanhttp v0.5.2
go: downloading github.com/hashicorp/go-rootcerts v1.0.2
go: downloading github.com/mitchellh/mapstructure v1.4.1
go: downloading github.com/gdamore/encoding v1.0.0
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading golang.org/x/term v0.0.0-20220722155259-a9ba230a4035
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/rivo/uniseg v0.4.2
# github.com/rivo/uniseg
/go/pkg/mod/github.com/rivo/uniseg@v0.4.2/properties.go:137:6: missing function body
/go/pkg/mod/github.com/rivo/uniseg@v0.4.2/properties.go:137:20: syntax error: unexpected [, expecting (
note: module requires Go 1.18
The command '/bin/sh -c go build -o damon ./cmd/damon' returned a non-zero code: 2

```

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,7 +7,7 @@
 
 # devbuild compiles the binary
 # -----------------------------------
-FROM golang:1.17 AS devbuild
+FROM golang:1.19.1 AS devbuild
 
 # Disable CGO to make sure we build static binaries
 ENV CGO_ENABLED=0

